common/flatpak-dir.c

```diff
@@ -14615,11 +14615,13 @@ flatpak_dir_get_locale_subpaths (FlatpakDir *self)
   /* Convert languages to paths */
   for (i = 0; subpaths[i] != NULL; i++)
     {
-      char *lang = subpaths[i];
```

Comment:
I don't think this change is right (unless you actually plan to change how locale extensions are split up).

Instead, flatpak_dir_get_default_locale_languages needs to be changed to keep doing what it is promising to do: return a list of languages. With the new semantics of extra-languages, that means it now needs to strip the region parts from the locale identifiers it runs across.

And the same goes for all other places where the extra-languages key is used. There is at least one other, in the system-helper code.

Reply:
I didn't get the part where you pointed the system-helper code - doesn't the system-helper interfaces the issues that the `app/flatpak-builtins-config.c` may come across?

Reply:
I just did: 

```
$ git grep -l extra-languages
app/flatpak-builtins-config.c
common/flatpak-dir.c
doc/flatpak-config.xml
system-helper/flatpak-system-helper.c
tests/testlibrary.c
```

Reply:
Looks like the system-helper just passes along the config value to `flatpak_dir_set_config` so I don't think anything needs changing there.